### PR TITLE
test: added E2E tests for StarSearch logged out experience

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Adds end to end (E2E) tests for the StarSearch logged out experience. Writing the tests also uncovered some accesibility issues with StarSearch that have been addressed.

## Related Tickets & Documents

Relatest to https://github.com/open-sauced/engineering/issues/294

## Mobile & Desktop Screenshots/Recordings

N/A

## Steps to QA

E2E tests go green

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: components/StarSearch/LoginModal.tsx [hunk 1]
```diff
@@ -51,7 +51,12 @@ const StarSearchLoginModal = ({ isOpen, onClose }: StarSearchLoginModalProps) =>
     </Drawer>
   ) : (
     <Dialog open={isOpen}>
-      <DialogContent autoStyle={false} onEscapeKeyDown={onCloseModal} onPointerDownOutside={onCloseModal}>
+      <DialogContent
+        autoStyle={false}
+        onEscapeKeyDown={onCloseModal}
+        onPointerDownOutside={onCloseModal}
+        aria-label="Login to try StarSearch"
```
Comment (nickytonline) [inline components/StarSearch/LoginModal.tsx:58]: The dialog didn't have a label.

File: pages/star-search/index.tsx [hunk 2]
```diff
@@ -627,6 +627,7 @@ export default function StarSearchPage({ userId, ogImageUrl }: StarSearchPagePro
                 }}
               />
               <button type="submit" disabled={isRunning} className="bg-white p-2 rounded-r-lg">
+                <span className="sr-only">Submit your question to StarSearch</span>
```
Comment (nickytonline) [inline pages/star-search/index.tsx:630]: The button didn't have a label.

File: pages/star-search/index.tsx [hunk 3]
```diff
@@ -669,16 +670,21 @@ function SuggestionBoxes({
       } gap-2 lg:gap-4 w-full max-w-3xl`}
     >
       {suggestions.map((suggestion, i) => (
-        <button key={i} onClick={() => addPromptInput(suggestion.prompt)}>
-          <Card
-            className={`${
-              isHorizontal ? "w-[30rem] snap-start" : "w-full h-fit"
-            } shadow-md border-none text-start !p-6 text-slate-600`}
+        <Card key={i} className="!p-6 shadow-md border-none text-slate-600">
+          <button
```
Comment (nickytonline) [inline pages/star-search/index.tsx:674]: The buttons for suggestions didn't have labels. The style of the suggestions is still the same, but now they are more accessible.

**Review by brandonroberts — APPROVED** [review]

**Review by BekahHW — APPROVED** [review]

**Pull request merged**